RSPlayPauseButtonDemo/RSPlayPauseButton/RSPlayPauseButton.m

```diff
@@ -229,17 +229,19 @@ - (CGSize)sizeThatFits:(CGSize)size
 - (void)layoutSubviews
 {
     [super layoutSubviews];
-    
-    if (!self.borderShapeLayer) {
-        self.borderShapeLayer = [[CAShapeLayer alloc] init];
-        // Adjust for line width.
-        CGRect borderRect = CGRectInset(self.bounds, ceil(kBorderWidth / 2), ceil(kBorderWidth / 2));
-        self.borderShapeLayer.path = [UIBezierPath bezierPathWithOvalInRect:borderRect].CGPath;
-        self.borderShapeLayer.lineWidth = kBorderWidth;
-        self.borderShapeLayer.fillColor = [UIColor clearColor].CGColor;
-        [self.layer addSublayer:self.borderShapeLayer];
+
+    if (self.hasBorderCircle) {
```

Comment:
Instead of nesting if's, the expression could be combined.

Reply:
Yeah you are right, when I initially authored this change I wanted my added code to have the least impact possible. That unfortunately meant duplicating a portion of your code and introducing a nested `if`.